app/packages/core/src/components/Actions/Similar.tsx

```diff
@@ -67,39 +86,52 @@ const useSortBySimilarity = (close) => {
       async (parameters: fos.State.SortBySimilarityParameters) => {
         set(fos.similaritySorting, true);
 
-        const queryIds = await getQueryIds(snapshot, parameters.brainKey);
+        const queryIds = parameters.query
+          ? null
+          : await getQueryIds(snapshot, parameters.brainKey);
         const view = await snapshot.getPromise(fos.view);
         const subscription = await snapshot.getPromise(fos.stateSubscription);
-        const data: fos.StateUpdate = await getFetchFunction()(
-          "POST",
-          "/sort",
-          {
-            dataset: await snapshot.getPromise(fos.datasetName),
-            extended: toSnakeCase({
-              ...parameters,
-              queryIds,
-            }),
-            filters: await snapshot.getPromise(fos.filters),
-            subscription,
-            view,
-          }
-        );
 
-        update(({ set }) => {
-          set(fos.similarityParameters, {
-            ...parameters,
-            queryIds,
+        const { query, ...commonParams } = parameters;
+
+        const combinedParameters = {
+          ...commonParams,
+        };
+
+        combinedParameters["query"] = query ?? queryIds;
+
+        try {
+          const data: fos.StateUpdate = await getFetchFunction()(
+            "POST",
+            "/sort",
+            {
+              dataset: await snapshot.getPromise(fos.datasetName),
+              view,
+              subscription,
+              filters: await snapshot.getPromise(fos.filters),
+              extended: toSnakeCase(combinedParameters),
+            }
+          );
+
+          // update selectedSamples atom to new set.
+          // modifying data, otherwise useStateUpdate will set the selectedSamples again
+          data.state.selected = [];
```

Comment:
This is not allowed. You cannot mutate state on the client like this. I think you need to merge `develop` to get some recent changes I made

Reply:
This is not needed, either. I am stripping it.